src/sage/calculus/functions.py

```diff
@@ -106,7 +106,7 @@ def row(n):
         return matrix([row(r) for r in range(len(fs))]).determinant()
 
 
-def jacobian(functions, variables):
+def jacobian(functions, variables, hessian=False, sparse=False, rank=False):
```

Comment:
I think that `hessian` should be its own function unless you have a good reason for it not to be separate.

I don't think we need the `rank` option since the returns matrix already has a `.rank()` method that can be called on it.